var/spack/repos/builtin/packages/glib/package.py

```diff
@@ -42,6 +42,7 @@ class Glib(AutotoolsPackage):
     version('2.42.1', '89c4119e50e767d3532158605ee9121a')
 
     variant('libmount', default=False, description='Build with libmount support')
+    variant('dtrace', default=True, description='Build with dtrace support')
```

Comment:
Since `glib+dtrace` will fail if `dtrace` is present on the system, I think this should be disabled by default and enabled only when really needed. @alalazo @luigi-calori any thoughts?

Reply:
I have no idea of what dtrace is used for. 
I have tried to stick as much as possible with previous behavior.
As it seems that dtrace is on by default, in glib, I set the default variant accordingly.
I' m currently trying to see if  #7157 can be a substitute of this PR

Reply:
As far as I can see `dtrace` starts with:
```bash
#!/usr/bin/python
# vim: et sta sts=4 sw=4 ts=8
...
```
At install time, if `dtrace` is present, we could put in path a "fake" `dtrace` that does:
```bash
#!/usr/env bash
python /usr/bin/dtrace
```
so that the `python` used is consistent with what is in `PYTHONHOME`. Otherwise I think #7157 can't do much for a script that hardcodes which interpreter to use.

Reply:
@luigi-calori I generally agree with this:

> I have tried to stick as much as possible with previous behavior.

...but in this case, default means *broken by deafult* :)

@alalazo Just like @luigi-calori I have no idea about what does *`dtrace` support* means in `glib`. We can try this *dtrace monkeypatching* or leave it disabled (with some warning when `+dtrace` maybe), no preference at this point.

Reply:
I can confirm that , unfortunately, #7157 does not fix #6965
In this PR I have undefined PYTHONHOME var ( hopefully only at build time) as 
build_env.unset('PYTHONHOME')
It seems that is this setting that harm any system python script that explicitly point to system python.
Could it make sense?

Reply:
I think that the problem of setting PYTHONHOME to spack installed python ( happens when depending on python) can affect any package that rely on any python script that point 
explicitly to system python
I have no idea why this variable is used but, in this particular case, where a build rely on a system provided python script, it seem to fix the issue.
It could happen with another script apart from dtrace, so disabling d